src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -210,19 +210,22 @@ def require_package
     required_parameters :package
     params[:rev], params[:package] = params[:pkgrev].split('-', 2) if params[:pkgrev]
     @project ||= params[:project]
-    if params[:package].present?
-      begin
-        @package = Package.get_by_project_and_name(@project.to_param, params[:package],
-                                                   use_source: false, follow_project_links: true, follow_multibuild: true)
-      rescue APIError # why it's not found is of no concern :)
-      end
-    end
 
-    return if @package
+    return if params[:package].blank?
 
-    raise(ActiveRecord::RecordNotFound, 'Not Found') unless request.xhr?
+    begin
+      @package = Package.get_by_project_and_name(@project.to_param, params[:package],
+                                                 follow_project_links: true, follow_multibuild: true)
+    rescue APIError => error
+      if [Package::Errors::ReadSourceAccessError, Authenticator::AnonymousUser].include?(error.class)
+        flash[:error] = "You don't have access to the sources of this package: \"#{params[:package]}\""
+        redirect_back(fallback_location: project_show_path(@project))
+        return
+      end
 
-    render nothing: true, status: :not_found
+      raise(ActiveRecord::RecordNotFound, 'Not Found') unless request.xhr?
```

Comment:
if you add the ```before_action check_ajax``` as in ```render_dialog``` this check can go away, no?

Reply:
@vpereira, no because this is not only for ajax.

Reply:
So if this is resolved, then it's not a blocker anymore. Please have a look again @vpereira if something else has to be changed.